src/iop/useless.c

```diff
@@ -87,18 +113,57 @@ int default_colorspace(dt_iop_module_t *self, dt_dev_pixelpipe_t *pipe, dt_dev_p
   return iop_cs_Lab;
 }
 
+// whenever new fields are added to (or removed from) dt_iop_..._params_t or when their meaning
+// changes, a translation from the old to the new version is added here.
+// A verbatim copy of the old struct definition should be copied into the routine with a _v?_t ending.
+// Since this will get very little future testing (because few developers still have very
+// old versions lying around) existing code should be changed as little as possible, if at all.
+//
+// upgrading from an older version than the previous one should always go through all in between versions
+// (unless there was a bug) so that the end result will always be the same.
+//
+// Be careful with changes to structs that are included in _params_t
+//
+// individually copy each existing field that is still in the new version. This is robust even if reordered.
+// If only new fields were added at the end, one call can be used:
+//   memcpy(n, o, sizeof(o));
+//
+// hardcode the default values for new fields that were added, rather than referring to default_params;
+// in future, the field may not exist anymore or the default may change. the best default for a new version
+// to replicate a previous version might not be the optimal default for a fresh image.
+//
```

Comment:
Again clarify that the following is an example and comment on what it's doing.

Reply:
well, it's not doing anything useful, as promised ;-)

upgrading the example to do something mildly interesting/sensible would be a bigger job.

I probably made it less sensible by adding some random fields just to demonstrate the two version upgrade process...